src/modulefinder/sentry_modulefinder_linux.c

```diff
@@ -162,8 +165,63 @@ get_code_id_from_elf(void *base, size_t *size_out)
     return NULL;
 }
 
-static sentry_value_t
-module_to_value(const sentry_module_t *module)
+static sentry_uuid_t
+get_code_id_from_text_fallback(void *base)
+{
+    const uint8_t *text = NULL;
+    size_t text_size = 0;
+
+    const uint8_t *addr = base;
+    // iterate over all the program headers, for 32/64 bit separately
+    const unsigned char *e_ident = addr;
+    if (e_ident[EI_CLASS] == ELFCLASS64) {
+        const Elf64_Ehdr *elf = base;
+        const Elf64_Shdr *strheader = (const Elf64_Shdr *)(addr + elf->e_shoff
+            + elf->e_shentsize * elf->e_shstrndx);
+
+        const char *names = (const char *)(addr + strheader->sh_offset);
+        for (int i = 0; i < elf->e_shnum; i++) {
+            const Elf64_Shdr *header = (const Elf64_Shdr *)(addr + elf->e_shoff
+                + elf->e_shentsize * i);
+            const char *name = names + 1 + header->sh_name;
+            if (header->sh_type == SHT_PROGBITS && strcmp(name, "text") == 0) {
```

Comment:
Adding `+1` to the name and then comparing to `"text"` seems a bit obscure. Could you compare to `".text"` directly?